pages/news/hokiport.md

```diff
@@ -0,0 +1,48 @@
+
+# Kernel sources
+
+The challenge began with Fossil Group not having the kernel sources available for download.
+A community member (Zoey) and I contacted Fossil Group separately to inquire for the kernel sources as they legally should because the Linux kernel is licensed under GPL-2.0.
+I didn't think this would be particularly hard since all previous Fossil watches have the kernel sources hosted by [Google](https://android.googlesource.com/kernel/msm/+refs).
+But it took a while for Fossil Group to actually release the kernel sources.
+While Fossil Group mentioned things like: "Fossil supports open source and complies with open source licenses." and "Our Wear OS by Google based smartwatches are using the Linux Kernel which is licensed under GPL." while also linking Google kernel sources they couldn't help directly when it came to the Fossil Gen 6.
+As I contacted Fossil on separate occasions about this I was ensured that this issue was raised internally and I would get an update by e-mail. It took a couple of tries but eventually a case number was created and the conversation continued via e-mail as opposed to their online chat.
+After some back and forth via e-mail kindly asking for the kernel sources they eventually escalated this to the engineering team. This was around the end of 2022. I decided to give Fossil Group some time to get the kernel sources ready. And contacted them again at the beginning of April. At which point they responded to wait a little longer. Two days later the kernel sources were [released](https://github.com/fossil-engineering/kernel-msm-fossil-cw)!
+
+This would make it possible to start a new port.
+
+# The port
+
+The watch originally launched with Android 9 based WearOS 2.0 which someone on XDA has posted for [download](https://forum.xda-developers.com/t/need-fossil-gen-6-ota-url.4387157/post-86792599).
+This turned out to make an initial port relatively straightforward with some exceptions of course 😅.
+
+Getting the kernel to compile was a bit challenging as the vendor kernel doesn't actually provide a [device tree](https://github.com/fossil-engineering/kernel-msm-fossil-cw/blob/fossil-android-msm-hoki-lw1.2-4.14/arch/arm64/boot/dts/qcom/Makefile#L2). The device tree was recreated by dumping the flash memory using [this](https://forum.xda-developers.com/t/twrp-twrp-for-fossil-gen-6-aka-hoki.4474863/) TWRP image followed by decompiling the Device tree Blob and packaging that with the kernel source. There was also a function missing. Once this was fixed the kernel started booting!
+
+Relatively few other patches were required to get the AsteroidOS UI up and running. It mostly consisted of adapting `beluga` specific changes such that they would work with `hoki`.
+
+Obviously this port isn't perfect so follow our [issues]((https://github.com/AsteroidOS/meta-smartwatch/issues?q=is%3Aissue+is%3Aopen+label%3Ahoki)) and check the website for the latest support status updates!
```

Comment:
This is the same as the third paragraph. Maybe we should only keep one or the other ?